version: "3.1"
services:
  mysql:
    container_name: mysql
    image: mysql:8.0.27
    command: --default-authentication-plugin=mysql_native_password
    ports:
      - 8306:3306
    volumes:
      - /var/lib/mysql:/var/lib/mysql
      - /etc/mysql/mysql.conf.d:/etc/mysql/conf.d
      - /etc/init.d/mysql:/etc/init.d/mysql
      - /var/log/mysql/error.log:/var/log/mysql/error.log
      - /var/run/mysqld/mysqld.sock:/var/run/mysqld/mysql.sock
    environment:
      MYSQL_DATABASE: "test11_client_jenkins"
      MYSQL_USER: "root"
      MYSQL_PASSWORD: "Zerkalo82"
#      MYSQL_ROOT_USER: "root"
      MYSQL_ROOT_PASSWORD: "Zerkalo82"
#      MYSQL_ROOT_HOST: '%'
    restart: "no"
    networks:
      - test11

  test11_client_jenkins:
    container_name: container_client_app
    image: solnce52004/test11_client_jenkins:latest
#    network_mode: "host" - "host" network_mode is incompatible with port_bindings
    ports:
      - 5443:4443
    volumes:
      - .:/test11_client_jenkins
      - ./logs:/test11_client_jenkins/logs
#    environment:
#      DB_URL: jdbc:mysql://mysql:3306/test11_client_jenkins
    depends_on:
      - mysql
    restart: "no"
    networks:
      - test11

networks:
  test11:
    external: true
    name: test11
    driver: bridge

